### PR TITLE
Block requests have higher priority

### PR DESCRIPTION
**Pull request by LuKks**

### CODE REVIEW

File: lib/receiver-queue.js [hunk 1]
```diff
@@ -3,19 +3,26 @@ const FIFO = require('fast-fifo')
 module.exports = class ReceiverQueue {
   constructor () {
     this.queue = new FIFO()
+    this.priority = new FIFO()
     this.requests = new Map()
     this.length = 0
```
Comment (LuKks) [inline lib/receiver-queue.js:8]: Looks equal to `requests.size` based on current usage

Comment (LuKks) [reply lib/receiver-queue.js:8]: Ah it's not the same, if two requests with same id are pushed

File: lib/receiver-queue.js [hunk 2]
```diff
@@ -3,25 +3,40 @@ const FIFO = require('fast-fifo')
...
   constructor () {
     this.queue = new FIFO()
+    this.priority = []
     this.requests = new Map()
     this.length = 0
   }
 
   push (req) {
-    this.queue.push(req)
+    if (req.priority > 0) this.priority.push(req)
```
Comment (mafintosh) [inline lib/receiver-queue.js:12]: can you add a TODO above this line to use a heap at some point if we wanna support multiple prios

**Comment by mafintosh** [comment]
Looks good!

**Pull request merged**